YourEasyRent/Controllers/ProductController.cs

```diff
@@ -17,77 +17,139 @@ public class ProductController : ControllerBase
+
+            catch (Exception ex) 
+            {
+                Console.WriteLine($"[Get] : {ex.Message}");
+                return StatusCode(404, "Unable to process the request. Product not found");
             }
-            return Ok(product);
         }
+
         [Route("[action]/{brand}", Name = "GetProductByBrand")]
         [HttpGet]
         public async Task<ActionResult<Product>> GetProductByBrand(string brand)
         {
-            var brandProduct = await _repository.GetByBrand(brand);
-            if (brandProduct == null)
+            try
+            {
+                var brandProduct = await _repository.GetByBrand(brand);
+                if (brandProduct == null)
+                {
+                    return NotFound();
+                }
+                return Ok(brandProduct);
+            }
+
+            catch (Exception ex) 
             {
-                return NotFound();
+                Console.WriteLine($"[GetByBrand] : {ex.Message}");
+                return StatusCode(404, "Unable to process the request. Brand not found");
             }
-            return Ok(brandProduct);
         }
+
         [Route("[action]/{name}", Name = "GetProductByName")]
         [HttpGet]
         public async Task<ActionResult<Product>> GetProductByName(string name)
         {
-            var nameProduct = await _repository.GetByName(name);
-            if (nameProduct == null)
+            try
             {
-                return NotFound();
+                var nameProduct = await _repository.GetByName(name);
+                if (nameProduct == null)
+                {
+                    return NotFound();
+                }
+                return Ok(nameProduct);
+            }
+            catch (Exception ex) 
+            {
+                Console.WriteLine($"[GetByName] : {ex.Message}");
+                return StatusCode(404, "Unable to process the request. Name not found");
             }
-            return Ok(nameProduct);
         }
 
         [HttpPost]
         public async Task<IActionResult> Post([FromBody]Product newProduct)
         {
-            await _repository.Create(newProduct);
-            return CreatedAtAction(nameof(Get), new { id = newProduct.Id }, newProduct);
-
+            try
+            {
+                await _repository.Create(newProduct);
+                return CreatedAtAction(nameof(Get), new { id = newProduct.Id }, newProduct);
+            }
+            catch (Exception ex) 
+            {
+                Console.WriteLine($"[Create] : {ex.Message}");
+                return StatusCode(500, "Failed to create user.");
+            }
         }
         
 
 
         [HttpPut("{id:length(24)}")]
         public async Task<IActionResult> UpdateProduct(string id, Product updateProduct)
         {
-            var result = await _repository.Update(id, updateProduct);
+            try
+            {
+
+                var result = await _repository.Update(id, updateProduct);
```

Comment:
If you want to separate `Id` you also need a separate model for `updateProduct`. 
Does this action work? I think it will fail because you will try to insert a product with **NULL** id